### ISSUE TITLE
ASN.1 GeneralizedTime parsing fails if there's no optional millisecond part

### ISSUE THREAD

**Issue by RockyYara** [outfoxx/PotentCodables#13]
![image](https://user-images.githubusercontent.com/29794800/132215848-31586f41-339c-4bf9-a362-77f6ed33162c.png)

My solution which solves the issue:

PotentCodables/Sources/PotentASN1/ASN1Reader.swift

```
    case .generalizedTime:
      let string = try parseString(&itemBuffer, tag: tag, encoding: .ascii)
      
      let dateFormatter: DateFormatter
      
      if string.contains(".") {
        dateFormatter = generalizedFormatter
      } else {
        dateFormatter = generalizedFormatterWithoutMillis
      }

      return .generalizedTime(dateFormatter.date(from: string)!)

private let generalizedFormatterWithoutMillis: DateFormatter = {
  let formatter = DateFormatter()
  formatter.calendar = Calendar(identifier: .iso8601)
  formatter.locale = Locale(identifier: "en_US_POSIX")
  formatter.timeZone = TimeZone(secondsFromGMT: 0)
  formatter.dateFormat = "yyyyMMddHHmmssXXXXX"
  return formatter
}()

```

I can push and make a PR if it's OK.

**Comment by kdubb** [comment]
@RockyYara There is a fix in #14 that fixes this correctly. It's a bit more complicated because DateFormatter lacks optional fields. Let me know if this works for you and I'll merge and cut a new release.

**Comment by kdubb** [comment]
@RockyYara #14 has morphed into a more formal fix with a new type `ZonedDate` added to fully support the GeneralizedTime value. It will be released under a `2.0` version as it changes the public API incompatibility.

**Comment by RockyYara** [comment]
> Let me know if this works for you and I'll merge and cut a new release.

@kdubb Thank you for your work! Yes, your code from #14 looked the way it fixes my issue. 

Current master works.
